login/login.go

```diff
@@ -0,0 +1,120 @@
+package login
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/usso"
+	"golang.org/x/crypto/ssh/terminal"
+	"gopkg.in/errgo.v1"
+)
+
+type TokenGetter interface {
```

Comment:
Why does this even need to be exported? It's not in the public API AFAICS and it's somewhat confusing when juxtaposed with TokenStore.